.github/workflows/test.yml

```diff
@@ -1,32 +1,54 @@
+
 name: build
+
 on:
+  pull_request:
+    types: [ opened, synchronize ]
   push:
-    branches: '*'
+    branches: [ master ]
+
```

Comment:
From my understanding, this will run the tests for any feature branch, is that correct? This is really useful for the tests to run on branches for every push.

Reply:
Do we need to run on each push? I thought just on PRs would be sufficient. 

In my view, a developer can (and should) run builds and test code locally before pushing it, and certainly before making a PR. The checks in this workflow are meant to act as a safety net to catch anything that might have been missed before it goes into master (and potentially into a release).